pkg/apis/pipeline/v1beta1/param_types.go

```diff
@@ -141,17 +141,40 @@ type ArrayOrString struct {
 
 // UnmarshalJSON implements the json.Unmarshaller interface.
 func (arrayOrString *ArrayOrString) UnmarshalJSON(value []byte) error {
+	// ArrayOrString is used for Results Value as well, the results can be any kind of
+	// data so we need to check if it is empty.
+	if len(value) == 0 {
+		arrayOrString.Type = ParamTypeString
+		return nil
+	}
```

Comment:
Would this impact parameters with no values? is it covered by alternate validation to make sure a required parameter is specified - please confirm that the parameter validation has a test to check for no empty value for required params.

Reply:
Thanks! Could you point me to a case where parameters has no values?
Correct if I'm wrong, if the parameters has no values then it will be captured by validation check 
For example, this won't work.  And won't reach this function. This case is special for results since user can write custom data to result path and it could be empty

```
apiVersion: tekton.dev/v1beta1
kind: TaskRun
metadata:
  name: validation-1
spec:
  taskSpec:
    params:
      - name: foo
        type: string
        default:
    steps:
      - name: echo-params
        image: bash
        args:
          [
            "echo",
           "$(params.foo)"]
```

Reply:
So validation will check if the param is empty:
https://github.com/tektoncd/pipeline/blob/7aef957effb55a519bba97c9d2c36a28482b35d4/pkg/reconciler/taskrun/validate_resources.go#L110